docs/blog/2019-01-29-themes-update-child-theming-and-component-shadowing/index.md

```diff
@@ -0,0 +1,131 @@
+now use the `__experimentalThemes` gatsby-config key in a theme as well as your
+sites.
+
+This change is being made to make it easier for theme authors to produce themes
+that rely on complex behavior while enabling users who have different skill sets
+to consume and modify those themes. Theme authors will benefit from the ability
+to abstract logic, UI elements, and other Gatsby idioms into multiple packages
+which can be consumed as a single theme. Child themes (and themes as a whole)
+will improve the ability of people with different skill sets to create
+performant, beautiful sites. A few that I had in mind while building out child
+theming are:
+
+- A designer who wants to change a set of design tokens to create a whole new
+  look for a theme
+- Someone who is more familiar with HTML and CSS than React and GraphQL wants to
+  create a set of child blog themes
+- Someone familiar with GraphQL wants to enable new functionality by writing
+  their own queries for page templates
+
+I also want child themes to enable an ecosystem to build on top of and compose
+with each other. The 80% use case for a blog should be capable of being built on
+a similar data model encapsulated in a parent theme so that users can swap out
+child themes to try out new sites or new looks. This will take some more effort
+to achieve, so more on that at a later date.
+
+## A Concrete Example
+
+Taking advantage of child theming requires some familiarity with theming itself
+to be used effectively. If you're already experienced with themes and want to
+see how one possible way to upgrade your themes to use this feature, the
+corresponding PR in the [gatsby-theme-examples
+repo](https://github.com/ChristopherBiscardi/gatsby-theme-examples) has [also
+been
+merged](https://github.com/ChristopherBiscardi/gatsby-theme-examples/pull/13).
+
+Let's say you have a theme that contains a blog data model and some logic
+relating to how to render blog posts at specific pages, we'll call it
+`gatsby-theme-blog-core`. This theme renders JSON representations of blog posts
+at the relevant page locations instead of full UI because it's only meant to
+encapsulate the data model. We'll skip going over page creation logic in
+`gatsby-node.js` and instead focus on the following possible `gatsby-config.js`
+for the core theme.
+
+```js
+module.exports = {
+  plugins: [`gatsby-mdx`],
+}
+```
+
+Then we have another theme that handles applying a UI layer using to the core
+data model, `gatsby-theme-blog`. This theme is a child theme of
+`gatsby-theme-blog-core` and shadows the core theme's modules to render
+beautiful posts on each blog using any styling technology we want, like
+[Emotion](https://emotion.sh/) or [Sass](https://sass-lang.com/). Again, since
+this is a post on child theming itself we'll skip looking at how shadowing is
+applied to the parent and focus on one possible child theme `gatsby-config.js`.
+
+```js
```

Comment:
Filename references would help to clarify these theme config sections.

Reply:
what do you mean by "filename references"?

Reply:
I wasn't quite sure if these configs lived in separate theme directories...is that the case?

Filename references came to mind based on that assumption, I just realized they're called "formatted titles" in the styleguide: https://www.gatsbyjs.org/docs/gatsby-style-guide/#code-formatting-titles